pkg/controller/servicediscovery/servicediscovery_controller.go

```diff
@@ -103,33 +105,34 @@ func (r *ReconcileServiceDiscovery) Reconcile(request reconcile.Request) (reconc
 		return reconcile.Result{}, err
 	}
 
-	// Create Lighthouse-agent
-	lighthouseAgent := newLighthouseAgent(instance)
-
-	// Set ServiceDiscovery instance as the owner and controller
-	if err := controllerutil.SetControllerReference(instance, lighthouseAgent, r.scheme); err != nil {
+	if err = r.reconcileLighthouseAgent(instance, reqLogger); err != nil {
```

Comment:
You mean to export the it from from submariner_controller.go and use it here ? Doesn't it add a dependency which is not ideally required?

Reply:
I mean reuse the code in `reconcileDaemonSet` instead of duplicating it, either refactor all or part of it into a utility function. If you refactor out just the `RetryOnConflict` code then it looks like it would only need `r.client` passed in extra. If you also refactor out `SetControllerReference`, then you also need `owner` and `r.scheme`. I'd prefer the former.  So the signature would look like:

`func reconcileDaemonSet(c client.Client, daemonSet *appsv1.DaemonSet,  reqLogger logr.Logger) error`

It could go in a separate say _helpers.go_ file but I think it would be fine in _submariner_controller.go_.

Reply:
I just realized this controller is in a separate package. In that case then I'd say put it in _pkg/controller/helpers.go_.

Reply:
Done.